src/zero/once.clj

```diff
@@ -1,77 +1,44 @@
 (ns zero.once
-  "Low-level vars to define exactly once and auth related junk."
-  (:require [clojure.data.json  :as json]
-            [zero.util          :as util]
-            [zero.environments  :as env]
-            [zero.zero          :as zero])
-  (:import [com.google.auth.oauth2 UserCredentials]
-           [java.net URI]
-           (java.io FileInputStream)
-           (com.google.auth.oauth2 GoogleCredentials)))
-
-;; https://developers.google.com/identity/protocols/OAuth2InstalledApp#refresh
-;;
-(defn new-user-credentials
-  "NIL or new UserCredentials for call. "
+  "Manage some credentials."
+  (:require [clojure.data.json :as json]
+            [clojure.java.io   :as io]
+            [clojure.string    :as str]
+            [zero.debug]
+            [zero.environments :as env]
+            [zero.util         :as util]
+            [zero.zero         :as zero])
+  (:import [com.google.auth.oauth2 ServiceAccountCredentials]))
+
+(defn authorization-header-with-bearer-token
+  "An Authorization header with a Bearer TOKEN."
+  [token]
+  {"Authorization" (str/join \space ["Bearer" token])})
+
+(defn service-account-token
+  "Throw or return a bearer token for the service account SA."
+  [{:keys [file vault] :as sa}]
+  (-> (cond file  (io/file file)
+            vault (-> vault util/vault-secrets json/write-str .getBytes)
```

Comment:
If I understand this correctly, in the env file, only one of `file` and `vault` should be set to true when we create new envs?

Reply:
Maybe neither. It depends, but it will prefer :file for no good reason.

Reply:
I don't like this solution much. We can discuss what else I tried and maybe find something better.